### ISSUE TITLE
JAWS misspeaks label text if inputs inside a container that has tabindex="-1" applied.

### ISSUE THREAD

**Issue by browncow2020** [FreedomScientific/VFO-standards-support#152]
### Summary

If form inputs are inside of a container which has had tabindex="-1" applied to allow it to serve as a skip link point of focus, when focus is placed on one of the inputs, JAWS will announce other text not associated with the input that is also in the container with tabindex="-1".

### Expected result

Expect JAWs to just speak the text programmatically associated with the input (i.e.: its label)

### Actual result

JAWs speaks surrounding text that is not explicitly, programmatically associated with the input in addition to the text that is.

### Example

```
<button id="btn1">Button 1</button>
<div id="mainContent" tabindex="-1"> <!-- tabindex in place to allow a skip link to jump to and place keyboard focus at this point -->
	<div id="sibling1">I'm Sibling one</div>
	<div id="sibling2">
        <p>I'm Sibling two</p>
        <label for="input1">Input 1</label><input type="text" name="input1" id="input1"/><br>
        <label for="input2">Input 2</label><input type="text" name="input2" id="input2"/><br>
    </div>
</div>
<button id="btn2">Button 2</button>
```
When JAWS is running and the user tabs to the first input from the first button or shift+tabs to the second input from the second button, JAWS will speak “I’m Sibling one I’m Sibling two” along with the label text of the input that was tabbed to.

Removing the tabindex from the containing element will correct the issue but this behavior doesn't seem correct.

### Additional Information
JAWS v18.0.2324 running on Windows 7 + IE 11 v11.0.9600.19204 does NOT exhibit the problem.
JAWS v18.0.5038 running on Windows 7 + IE 11 v11.0.9600.19204 does exhibit the problem.

Using MS Edge does NOT exhibit the problem.

#### JAWS version and build number

v18.0.5038 (Windows 7)
v2019.1810.64 (Windows 10)
v2019.1812.49 (Windows 10)
 
#### Operating System and version

Windows 7
Windows 10

#### Browser and version:

IE11 v11.0.9600.19204

**Comment by scottaohara** [comment]
Hi @browncow2020 thank you for filing the issue.

Have you tested with Firefox and Chrome as well?  You indicated that this issue does not exist with Edge, but want to verify this is only an IE issue.

Thanks

**Comment by browncow2020** [comment]
The issue seems to only affect IE11.
I have confirmed that Firefox, Chrome, Edge all work as expected. Only IE11 seems to cause JAWS to announce all text inside the containing element that has the tabindex="-1" applied when a form input inside that container receives focus.

**Comment by stevefaulkner** [comment]
@browncow2020 this is an issue with how IE expresses "accessible elements" and how it calculates the [accessible name](https://www.w3.org/TR/accname-1.1/) for these elements.

Use of `tabindex=-1` adds the `div` to the accessibility tree with a `role=group` and an accessible name from the `div` content unless another accessible name is provided. Adding `aria-label=""` overrides name from content and provides an empty accessible name, thus alleviating the problem.

Before:
`<div tabindex="-1">`

After
`<div tabindex="-1" aria-label="">`

<img width="301" alt="accessibility tree showing accessible name for the div elements with tabindex=-1. The one without aria-label has an eccssible name consisting of the text content inside the div element." src="https://user-images.githubusercontent.com/835859/50561581-07004800-0d04-11e9-8bb0-a767914b0d97.PNG">

A [test case demonstrating the fix](https://s.codepen.io/stevef/debug/oJopWr)

**Comment by afercia** [comment]
Aside:

> a container which has had tabindex="-1" applied to allow it to serve as a skip link point of focus

Is `tabindex="-1"` still needed? As far as I can tell, WebKit used to have a bug which is now solved. All other browsers including Internet Explorer 11 seem they don't need the tabindex any longer?

We're facing the same issue in WordPress and got some precious insights from @stevefaulkner  feedback 🙂 See https://core.trac.wordpress.org/ticket/43154
